### ISSUE TITLE
Failures to load or instantiate a compiler/preprocess ICompileModule should be more verbose

### ISSUE THREAD

**Issue by Tragetaschen** [aspnet/dnx#1041]
[Currently](https://github.com/aspnet/KRuntime/blob/dev/src/Microsoft.Framework.Runtime.Roslyn/RoslynCompiler.cs#L157), Exceptions are put on the Trace. When something fails (version missmatch on the `!preprocess`-Assembly, failures to instantiate, etc.), the `ICompileModule`s might not have run or might have partly run. Nevertheless as per default, there is no indication in the log.

An example is the RazorPreCompileModule that might not have run during `pack` (I mixed different Roslyn versions -- bad idea)

**Comment by davidfowl** [comment]
This was done originally because it didn't run in all contexts. Now that everything is plumbed through, we're just going to let the exception happen.

**Comment by muratg** [comment]
Closing per @davidfowl comment.  @Tragetaschen please re-open if the exceptions are not easy to understand.

**Comment by Tragetaschen** [comment]
Ok, this (or something very similar) bit me again when moving from beta4 to beta5.

The change to `ICompileModule` was to remove the 'I's from both Before- and AfterCompileContext, so my compile modules needed to change. The net effect when running `dnu build` on a command line was several errors because my generated classes weren't present (and also a test `Console.WriteLine` didn't make it).

So I was in the nice situation where:
- No samples for `ICompileModule` were available
- No immediate error in Visual Studio (compiler/preprocess code isn't part of IntelliSense)
- All the usual errors in VS like in beta4, because ICompileModules are not run on the DTH
- No immediate error on the command line
That was no fun.

As a workaround, I copied the class from the compiler/preprocess folder to the root of the project and there I was able to see the actual errors.
